extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/DefaultHttpPermissionChecker.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.HttpBuildTimeConfig;
+import io.quarkus.vertx.http.runtime.PermissionSetConfig;
+import io.vertx.core.http.HttpServerRequest;
+
+/**
+ * permission checker that handles permissions defined in application.properties
+ */
+@ApplicationScoped
+public class DefaultHttpPermissionChecker implements HttpPermissionChecker {
```

Comment:
Would it make sense to have a property, for example in the `(Default)AuthConfig` to have this checker disabled when some more specific checker is only required, say some claim based OIDC checker ?

Reply:
This checker is enabled if role based permission checks have been added. If another checker is added as well then both will be used.